### PR TITLE
Fix GH#22058 and GH#22061: Fix issues with Copy lyrics to clipboard

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
* Fix GH#22058: Copy lyrics of all voices to clipboard
* Code cleanup
* Fix GH#22061: Don't skip last verse lyrics syllables in Copy Lyrics to Clipboard when not having a syllable in an earlier verse

Resolves: #22058 
Resolves: #22061

### CODE REVIEW

File: src/engraving/dom/score.cpp [hunk 1]
```diff
@@ -4926,40 +4923,36 @@ String Score::extractLyrics()
...
+                    if (l->syllabic() == LyricsSyllabic::SINGLE || l->syllabic() == LyricsSyllabic::END) {
+                        result += lyric + u" ";
+                    } else if (l->syllabic() == LyricsSyllabic::BEGIN || l->syllabic() == LyricsSyllabic::MIDDLE) {
+                        result += lyric;
                     }
                 }
             }
         }
-        if (found) {
-            result += u"\n\n";
```
Comment (Jojo-Schmitz) [inline src/engraving/dom/score.cpp:4961]: gets trimmed away furter down anyway

File: src/engraving/dom/score.cpp [hunk 2]
```diff
@@ -4882,40 +4882,38 @@ String Score::extractLyrics()
...
-                    ChordRest* cr = toChordRest(seg->element(static_cast<int>(track)));
+                    ChordRest* cr = toChordRest(seg->element(track));
                     if (!cr || cr->lyrics().empty()) {
                         continue;
                     }
                     if (cr->lyrics().size() > maxLyrics) {
                         maxLyrics = cr->lyrics().size();
                     }
-                    if (playCount >= int(cr->lyrics().size())) {
+                    if (playCount >= cr->lyrics().size() + 1) {
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/dom/score.cpp:4905]: No ;-)
Basically the idea way consistency with the change further down

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/dom/score.cpp:4905]: It seems needed though: play count 5, 5 verses and we're having the problem again
But `if (playCount > cr->lyrics().size()) {` would work here too (not firther down though).
So back to the consistency argument

[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/engraving/dom/score.cpp:4905]: I think I get it now. I do think `if (playCount > cr->lyrics().size()) {` is slightly more intuitive. 

And further down, why can't you do `if (lyricsNumber > cr->lyrics().size()) {`?

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/dom/score.cpp:4905]: Let my try it again, but ISTR that it didn't work

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/dom/score.cpp:4905]: Although I don't see why it wouldn't

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/dom/score.cpp:4905]: OK, seems I remembered wrongly, or, moree likely, tested wrongly.
So I'll make that change for both (and rebase, while at ut)

**Comment by zacjansheski** [comment]
Tested on MacOS 14, Windows 11, Ubuntu 22.04.3. Approved
https://github.com/musescore/MuseScore/issues/22058 FIXED
https://github.com/musescore/MuseScore/issues/22061 FIXED

**Pull request merged**